File: 100-Days-Series/Day 93/FindElementInRange.py

# Function to find number in a particular range.
def get_find_number_in_range(arr,sr,er,number):
    if sr<0 or er>=len(arr) or sr>er:
        return -1
    
    for i in range(sr,er+1,1):
        if arr[i]==number:
            return i
        
    return -1

# Let's Start Here
arr=[1,2,3,4,5,6,7,8,9,10]
sr=4
er=7
numberToFind=6
res=get_find_number_in_range(arr,sr,er,numberToFind)
if res!=-1:
    print('Found at Index ',res)
else:
    print('Not Found!!')

File: 100-Days-Series/Day 93/FindElementInRange.cpp
#include<iostream>
using namespace std;
// Function to find a number in a particular range.
int getFindNumberInRange(int arr[],int sr,int er,int size,int number){
    if(sr<0 || er>=size || sr>er){
        return -1;
    }
    for(int i=sr;i<=er;i++){
        if(arr[i]==number){
            return i;
        }
    }
    return -1;
}
// Let's Start Here
int main(){
    int arr[]={1,2,3,4,5,6,7,8,9,10};
    int size=sizeof(arr)/sizeof(arr[0]);
    int sr=4;
    int er=8;
    int numberToFind=6;
    int res=getFindNumberInRange(arr,sr,er,size,numberToFind);
    if(res!=-1){
        cout<<"\nFound at Index "<<res<<endl;
    }else{
        cout<<"\nNot Found!!"<<endl;    
    }
}